File: elixir/raindrops/raindrops.exs
defmodule Raindrops do
  def convert(num) do
    result = get_word(num, 3, "i") <> get_word(num, 5, "a") <> get_word(num, 7, "o")
    if result != "" do
      result
    else
      to_string num
    end
  end

  defp get_word(n, mod, letter), do: rem(n, mod) == 0 && "Pl" <> letter <> "ng" || ""
end

File: ecmascript/raindrops/raindrops.js
const maps = {3: "i", 5: "a", 7: "o"};

export default class Raindrops {
  convert(n) {
    let message = [3,5,7].map(factor => {
      if (n % factor === 0) {return `Pl${maps[factor]}ng`;}
      else { return ''; }
    }).join("")

    return message == '' ? n.toString() :  message;
  }
}
